Improve README (#38)
rootstrap/yaaf: README.md

If you want to learn more about Form Objects you can check out [these great articles](#links).

### Why YAAF?

- It is [64 lines long](https://github.com/rootstrap/yaaf/blob/master/lib/yaaf/form.rb#L64). As you can imagine, we did no magic in such a few lines of code, we just leveraged Rails modules in order to provide our form objects with a Rails-like behavior. You can review the code, it's easy to understand.

- It provides a similar API to `ActiveModel` models so you can treat them interchangeably.

> - You can customize it 100%. We promote having your own `ApplicationForm` which inherits from `YAAF::Form` and make the customizations you'd like for your app.

Reviewer: ```suggestion
- You can customize it 100%. We encourage you to have your own `ApplicationForm` which inherits from `YAAF::Form` and make the customizations you'd like for your app.
```

- It helps decoupling the frontend from the database. This is particularly important when using Rails as a JSON API with a frontend in React/Ember/Vue/Angular/you name it. If you were to use `accepts_nested_attributes_for` your frontend would need to know your database structure in order to build the request. With `YAAF` you can provide a the interface you think it's best.

- It easily supports nested models, collection of models and associated models. You have full control on their creation.

> - It helps you keep your models, views or controllers thin by providing a better place where to put business logic. In the end, this will raise the quality of your codebase and make it more maintainable and extensible.

Reviewer: ```suggestion
- It helps you keep your models, views or controllers thin by providing a better place where to put business logic. In the end, this will improve the quality of your codebase and make it easier to maintain and extend.
```

> - It helps you keep your models, views or controllers thin by providing a better place where to put business logic. In the end, this will raise the quality of your codebase and make it more maintainable and extensible.

Author: ```suggestion
- It helps you keep your models, views and controllers thin by providing a better place where to put business logic. In the end, this will raise the quality of your codebase and make it more maintainable and extensible.
```

> - It is an abstraction from production code. It has been working well for us, I'm confident it will work well for you too :)

Reviewer: <3
